modules/ROOT/nav.adoc

```diff
@@ -58,17 +58,19 @@
 ** xref:deploy:deployment-option/cloud/index.adoc[Cloud]
 *** xref:deploy:deployment-option/cloud/cloud-overview.adoc[Redpanda Cloud Overview]
 *** xref:deploy:deployment-option/cloud/whats-new-cloud.adoc[]
+*** xref:deploy:deployment-option/cloud/serverless.adoc[Serverless]
+*** xref:deploy:deployment-option/cloud/provision-a-byoc-cluster/index.adoc[BYOC]
```

Comment:
This is just to do a high level rearrangement of Cloud nav, right? So we're not yet looking into, say, also moving the VPC peering docs under BYOC and Dedicated (not sure if we were going to do that with the cloud migration either way)

Reply:
Yeah, not yet! Let's do that as part of the Cloud doc reorg.